.github/workflows/main.yml

```diff
@@ -0,0 +1,35 @@
+name: CI
+
+on: [push, pull_request] # on all pushes and PRs
+
+jobs:
+  ros1_bridge:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        distro: ["ros:rolling-ros1-bridge-focal", "ros:foxy-ros1-bridge-focal"]
+    container:
+      image: ${{ matrix.distro }}
+    env:
+      CCACHE_DIR: "/home/runner/.ccache"
+    steps:
+      - uses: actions/checkout@v2
+      - name: install ccache
+        run: |
+          sudo apt-get update -y
+          sudo apt-get -qq -y install ccache
+      - name: ccache cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ccache-${{ matrix.distro }}-${{github.run_id}}
+          restore-keys: |
+            ccache-${{ matrix.distro }}
```

Comment:
@nuclearsandwich @cottsay As we've had issues on the build farm related to ccache, could comment as to if you think there's any issue with this? Perhaps it's better to avoid caching altogether and live with longer CI times.

Reply:
Part of the reason we disabled ccache on the buildfarm is that the implementation leveraged a _shared_ ccache across packages, meaning that one package could poison the cache for everyone. Since this is a single ccache only for this package, I don't think that's a concern here. You may want to brush up on how to purge the cache and also determine the scope of the cache - for example, can someone open a PR with "bad" changes which poison the cache for normal runs.

I'd guess that the buildfarm concerns were rooted in security, and since the results of this job are discarded, it's less of a concern.

Reply:
> You may want to brush up on how to purge the cache and also determine the scope of the cache - for example, can someone open a PR with "bad" changes which poison the cache for normal runs.

I think the cache is not uploaded for runs that are not successful. But I can check this again.

Reply:
To add to @cottsay's answer. A threat faced by the build farm is that ccache itself has no security model so in addition to ccache placing objects in its own database as part of regular operations. Any script with access to the underlying ccache directory can also place potentially malicious objects manually which would be picked up as cache hits on subsequent builds. One reason this matters on the build farm is that the binary artifacts are the primary output of (the binary package jobs, at least) that farm. Here the produced artifacts are not propagated beyond the action and are only used for in-context testing. So an "attack" on the ccache database for this action's primary impact is a sort of denial-of-service attack by pushing a malformed cache object which would cause subsequent builds to fail all tests until the object was purged either by a change in the sources or a full cache purge.

All that's to say, I think ccache in this context is fine. I'd want to explore further if this action produced artifacts.

Reply:
Interesting!